### PR TITLE
Store eligibility override flag and reason in MG

### PR DESCRIPTION
**Pull request by scaustin34**
[TT6-185060667](https://www.pivotaltracker.com/story/show/185060667)

- adds member_determination model/entities
- sets and saves the memeber_determination fields as part of the override flow

### CODE REVIEW

File: app/operations/eligibilities/aptc_csr/apply_eligibility_overrides.rb [hunk 1]
```diff
@@ -15,15 +15,31 @@ def call(params)
...
+              if pregnancy_override?(thhm)
+                ped[:is_magi_medicaid] = true
+                ped[:member_determinations] << member_determ_for(thhm, :mitc_override_not_lawfully_present_pregnant)
+              end
+              if nineteen_to_twenty_one_override?(thhm)
+                ped[:is_magi_medicaid] = true
+                ped[:member_determinations] << member_determ_for(thhm, :mitc_override_not_lawfully_present_under_twenty_one)
+              end
+            elsif chip_ineligible_due_to_immigration_only?(thhm)
+              if pregnancy_override?(thhm)
```
Comment (erubinst) [inline app/operations/eligibilities/aptc_csr/apply_eligibility_overrides.rb]: I think if they are under 18 and pregnant, the pregnancy piece will be caught in the first if pregnancy_override? block (if you see the screenshot Michael attached yesterday, a 16 year old pregnant applicant will have the correct medicaid ineligibility reasons such that the medicaid_ineligible_due_to_immigration_only? function will return true) so I am not sure if the second catch here is needed.  But let me know if I am missing a piece of the logic.

File: spec/models/medicaid/aptc_household_member_spec.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+require 'rails_helper'
+
+RSpec.describe ::Medicaid::AptcHouseholdMember, type: :model, dbclean: :after_each do
+  context 'embeds_many :member_determinations' do
+    before do
+      @association = described_class.reflect_on_association(:member_determinations)
+    end
+
+    it "should return association's class as EmbedsMany" do
```
Comment (mdkaraman) [inline spec/models/medicaid/aptc_household_member_spec.rb]: These first few specs seem like they're testing the framework itself.  I think we could remove them and just keep the last one without losing meaningful test coverage.

File: app/operations/eligibilities/aptc_csr/apply_eligibility_overrides.rb (hunk 1 above)
Comment (scaustin34) [reply app/operations/eligibilities/aptc_csr/apply_eligibility_overrides.rb]: refactored logic to fix this

File: spec/models/medicaid/aptc_household_member_spec.rb (hunk 2 above)
Comment (scaustin34) [reply spec/models/medicaid/aptc_household_member_spec.rb]: removed!

**Review by mdkaraman — APPROVED** [review]
Looks good!  Allows the application to persist the new Member Determination data to the database and also populate the determination reasons for the Medicaid/CHIP override scenarios in the Application response payload.

**Review by erubinst — APPROVED** [review]
Looks good to me! Searching for prior member determinations before adding new reasons allows for multiple reasons to be stored and is good future proofing for if we start using the member determination fields in other locations and need to see if the applicant has any existing ones

**Pull request merged**